test/historical-query.spec.ts

```diff
@@ -40,7 +44,38 @@ let CONSEQUENCE: Consequence = {
 };
 
 describe("rules from AJO", () => {
-  it("supports historical condition for searchType ANY", () => {
+  const records = [
+    {
+      "iam.id":
+        "6cd5a8ed-e183-48b7-a0ef-657a4467df74#0477a309-6f63-4638-b729-ab51cf5dd3aa",
+      "iam.eventType": "display",
+      "iam.action": "",
+      someMagicalKey: 123,
+      futureKeys: "secret_abc_keys",
+      "ajo.id": 890,
+      firstTimestamp: 1609086720010,
+      timestamp: 1681321319855,
+    },
+    {
+      "iam.id":
+        "8cd5a8ed-e183-48b7-a0ef-657a4467df74#0477a309-6f63-4638-b729-ab51cf5dd3aa",
+      "iam.eventType": "interact",
+      "iam.action": "",
+      someMagicalKey: 123,
+      futureKeys: "secret_abc_keys",
+      "ajo.id": 890,
+      firstTimestamp: 1609086720010,
+      timestamp: 1681321319865,
+    },
+  ];
+
+  it("supports historical condition for searchType ANY", async () => {
+    const db = await setupFakeIndexedDBWithData(records);
```

Comment:
Yes, this is the right idea for testing.  Creating a fake IndexedDb with records just like you have done here.